### PR TITLE
added ability to fetch user by its email

### PR DESCRIPTION
**Pull request by Spartakovic**

### CODE REVIEW

**Comment by MartinsIrbe** [comment]
LGTM

File: src/main/java/org/atlasapi/application/users/v3/CacheBackedUserStore.java [hunk 1]
```diff
@@ -28,6 +29,12 @@ public CacheBackedUserStore(final UserStore delegate) {
                 return delegate.userForId(key);
             }
         });
+        this.emailCache = CacheBuilder.newBuilder().expireAfterAccess(10, TimeUnit.MINUTES).build(new CacheLoader<String, Optional<User>>() {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/users/v3/CacheBackedUserStore.java]: Readability. Add some newlines in the method chain

File: src/main/java/org/atlasapi/application/users/v3/CacheBackedUserStore.java [hunk 2]
```diff
@@ -45,6 +57,7 @@ public void store(User user) {
         delegate.store(user);
         userRefCache.invalidate(user.getUserRef());
         idCache.invalidate(user.getId());
+        emailCache.invalidate(user.getEmail());
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/users/v3/CacheBackedUserStore.java]: Why do we need to invalidate the cache here?

File: src/main/java/org/atlasapi/application/users/v3/MongoUserStore.java [hunk 3]
```diff
@@ -35,6 +36,13 @@ public MongoUserStore(DatabasedMongo mongo) {
     }
 
     @Override
+    public Optional<User> userForEmail(String email) {
+        BasicDBObject emailField = new BasicDBObject();
+        emailField.append(EMAIL, email);
+        return Optional.fromNullable(translator.fromDBObject(users.findOne(emailField)));
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/application/users/v3/MongoUserStore.java]: Is there an index on this field in Mongo?

**Pull request merged**